tests/semgrep/scripts/python_portability.py

```
@@ -0,0 +1,244 @@
+from tempfile import NamedTemporaryFile as named_file
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping, Sequence
+    from pathlib import Path
+
+import subprocess_utils as utils
+from subprocess_utils import run_git_command, run_git_command_with_input as git_input, run_safe_command
+
+
+def run_git_command_with_input(payload: str, argv: "Sequence[str]", options: "Mapping[str, Any]") -> None:
```

Ruff (0.16.3)

[warning] Missing docstring in public function

(D103)